src/wfs/src/main/java/org/geoserver/wfs/json/ComplexGeoJsonWriter.java

```diff
@@ -244,7 +246,24 @@ private void encodePropertiesByType(
                 encodeLinkedFeatures(descriptor, linkedFeatures);
             } else {
                 // no chained or linked features just encode each property
-                properties.forEach(this::encodeProperty);
+                String attributeName = descriptor.getName().getLocalPart();
```

Comment:
@fernandor777 look at he rest of the code and try to apply the same patterns :black_flag, more precisely in this case it means:
- isolate this behavior in an helper method, this else shoudl be a single line
- document properly, the comment above doesn't seem to match what you are doing
- if possible avoid this psychedelic Java 8 streams usage, it has performance personalities and it's not clear at all

Reply:
refactored